### ISSUE TITLE
Take several pictures

### ISSUE THREAD

**Issue by loran750** [City-of-Bloomington/uReport#360]
Hello

It would be wise to be able to take multiple pictures : either by the camera and either by selecting from the storage/disk.

regards

**Comment by inghamn** [comment]
Yes, It's been a long standing issue..  The current uReport software only supports posting pictures one at a time.  This means when the report is originally sent, the citizen can only post one photo.

City employees can add additional photos (one at a time), but that requires authentication - which the Open311 web api does not support.

**Comment by loran750** [comment]
Do you mean if on the front-end we are authenticated, we could post several photos ?
Perhaps getting around Open311 in order to post secondaries photos ?

**Comment by inghamn** [comment]
There are many problems to solve before this can happen; which is why this has lingered for so long.  Allowing only a single photo is a reasonable thing for most cases.

There is no mechanism, yet, for authenticated web service calls.
There is no support, yet, for citizens to have user accounts.

Adding additional photos means (editing an existing record in the system) - which requires authentication and permission to edit the record.  The easiest way forward is to alter the new ticket form to allow multiple photos in the initial post.

However, sending multiple photos in a single post brings up new issues. You start running up against HTTP Post request size limits.  The limits are there to prevent denial of service by bots posting lots of large files (which they try to do all the time).

**Comment by loran750** [comment]
If I choose this bloomington code, I will have to manage several photos.
Probably I would do it in 2 times in the same POST : 1st, creating the report with georeport V2 fields, 2nd, get the ID of the report just created then populate in another table or something like that.

**Comment by inghamn** [comment]
For us, our city staff typically add in additional photos using the CRM backend, while the report is being worked on.  That work does not happen using Open311.  They just use the web interface of this application.

It would still be useful to allow citizens to post more than one photo when reporting something, though.
